ckanext/canada/logic.py

```diff
@@ -460,3 +464,67 @@ def canada_job_list(up_func, context, data_dict):
         job['status'] = job_obj.get_status()
 
     return job_list
+
+
+@chained_action
+def canada_user_update(up_func, context, data_dict):
+    """
+    Checks if the sysadmin value is being changed.
+
+    We only want sysadmins from TBS to be able to do this.
+
+    We have to do it in this chained action, and not auth method
+    because sysadmins skip auth checks.
+    """
+    user = model.User.get(get_or_bust(data_dict, 'id'))
+
+    if not user:
+        raise ObjectNotFound(_('User not found'))
+
+    if data_dict.get('sysadmin', user.sysadmin) != user.sysadmin:
+        # the sysadmin value is trying to be changed.
+
+        # check if user has access to update users
+        check_access('user_update', context, data_dict)
+        contextual_user = context.get('auth_user_obj')
+        site_id = config.get('ckan.site_id')
+
+        # system user should be able to do anything still
+        if contextual_user.name == site_id:
+            return up_func(context, data_dict)
+
+        # check org membership to tbs-sct, confirm email, and sysadmin
+        tbs_membership = (model.Session.query(model.Group).
+                            filter(and_(model.Group.type == 'organization',
+                                        model.Group.state == 'active',
+                                        model.Group.name == 'tbs-sct')).
+                            join(model.Member, model.Member.group_id == model.Group.id).
+                            filter(and_(model.Member.table_id == contextual_user.id,
+                                        model.Member.table_name == 'user',
+                                        model.Member.state == 'active')))
+
+        # a bit crude, but don't want to be lax on sysadmin controllers.
+        # Note: system user does not have an email, use getattr
+        if not is_sysadmin(contextual_user.name) \
+                or not tbs_membership \
+                or not contextual_user.email \
+                or '@tbs-sct.gc.ca' not in contextual_user.email:
+                    raise NotAuthorized(
+                        _('User %s not authorized to modify sysadmins.')
+                        % contextual_user.name)
+
+        # cannot change your own sysadmin value
+        if user.name == contextual_user.name:
+            raise NotAuthorized(_('Cannot modify your own sysadmin privileges'))
+
+        # cannot change site user sysadmin value
+        if user.name == site_id:
+            raise NotAuthorized(_('Cannot modify sysadmin privileges for system user'))
+
+        # extra logging
+        if data_dict.get('sysadmin'):
+            log.info('%s promoted %s to a sysadmin', contextual_user.name, user.name)
+        else:
+            log.info('%s demoted %s from a sysadmin', contextual_user.name, user.name)
+
+    return up_func(context, data_dict)
```

Comment:
If some staging sysadmin is really determined they can still assign themselves to tbs then have full permission to do anything. This is a lot of code to prevent something unlikely and only possible on staging. If it's a concern why don't we remove the sysadmin permissions from non-tbs accounts on staging instead?

Reply:
Okay that makes sense. Did we still want to leave some of this code in? Such as preventing the modification of the system user? or do we not care about that too much?

Reply:
How about I just keep in the extra logging, and we can just remove all the other logic and just go with what Core does.

Reply:
@wardi okay, so I removed all of the TBS membership checks. Kept the no modify system user, and no modify your own privs. And kept in the extra logging

Will make a ticket to do an upstream contribution to the User Update schema later.